modules/db_virtual/dbase.c

```diff
@@ -512,3 +512,180 @@ int db_virtual_insert_update(const db_con_t* _h, const db_key_t* _k,
+                                                                                \
+				f->close(handle->con);                                          \
+				/* if failed before placing the fd in reactor                   \
+				 * we keep on */                                                \
+				if ((--__ah->cons_rem) == 0) {                                  \
+					LM_ERR("All databases failed!! No hope for you!\n");        \
+					return -1;                                                  \
+				}                                                               \
+                                                                                \
+				/* try next*/                                                   \
+				rc = -1;                                                        \
+				CURRCON(__ah)  = (CURRCON(__ah)+1)%p->size;                     \
+			} else {                                                            \
+				if (_resume_f)                                                  \
+					async_status = ASYNC_CHANGE_FD;                             \
+				set_update_flags(CURRCON(__ah), p);                             \
+				return rc;                                                      \
+			}                                                                   \
+		} else {                                                                \
+			LM_DBG("flags2 = %i\n", p->con_list[CURRCON(__ah)].flags);          \
+			if ((--__ah->cons_rem) == 0) {                                      \
+				LM_ERR("All databases failed!! No hope for you!\n");            \
+				return -1;                                                      \
+			}                                                                   \
+                                                                                \
+			/* try next*/                                                       \
+			rc = -1;                                                            \
+			CURRCON(__ah)  = (CURRCON(__ah)+1)%p->size;                         \
+		}                                                                       \
+		LM_DBG("curent_con = %i\n", CURRCON(__ah));                             \
+	} while ((__ah)->cons_rem); /* should never exit here */                    \
+                                                                                \
+	return rc;                                                                  \
+}while (0);
+
+
+int db_virtual_async_raw_query(db_con_t *_h, const str *_s, void **_data)
+{
+	handle_async_t* _ah;
+    handle_con_t * _handle;
+    handle_set_t * _p = (handle_set_t*)_h->tail;
+
+	if (_s->len > 16384) {
+		LM_ERR("query exceeds buffer size(16384)!\n");
+		return -1;
+	}
+
+	if ((_ah=pkg_malloc(sizeof(handle_async_t)+_s->len)) == NULL) {
+		LM_ERR("no more pkg\n");
+		return -1;
+	} else {
+		/* for RR next time the script shall trigger this,
+		 * other dest shall be called*/
```

Comment:
I literally had to read the code in order to understand the comment, rather than vice-verse!

Rule of thumb: when it comes to formulating comments, two things should come to mind: **well-written** vs. **badly-written** (language, acronyms, doc templates, coding style, etc.). **insightful** vs. **impeding** (usefulness of the comment itself: do you really have to *explain how round-robin works*, or simply that this code just does a *"round-robin increment"*?)